File: utilities/validation/go/validation_utils.go
package validation

import (
	"math"
	"regexp"
	"strings"
)

var emailPattern = regexp.MustCompile("^[^\\s@]+@[^\\s@]+\\.[^\\s@]+$")

func IsNonEmpty(value string) bool { return strings.TrimSpace(value) != "" }
func IsEmail(value string) bool { return emailPattern.MatchString(value) }
func IsWithinRange(value, minimum, maximum float64) bool {
	return !math.IsNaN(value) && !math.IsInf(value, 0) && value >= minimum && value <= maximum
}

File: utilities/validation/csharp/ValidationUtils.cs
using System.Text.RegularExpressions;

namespace PolyglotUtilities.Validation;

public static class ValidationUtils
{
    private static readonly Regex EmailPattern = new("^[^\\s@]+@[^\\s@]+\\.[^\\s@]+$", RegexOptions.Compiled);

    public static bool IsNonEmpty(string? value)
    {
        return !string.IsNullOrWhiteSpace(value);
    }

    public static bool IsEmail(string? value)
    {
        return value is not null && EmailPattern.IsMatch(value);
    }

    public static bool IsWithinRange(double value, double minimum, double maximum)
    {
        return double.IsFinite(value) && value >= minimum && value <= maximum;
    }
}
